master/internal/sproto/resources.go

```diff
@@ -224,12 +224,31 @@ func FromContainerFailureType(t aproto.FailureType) FailureType {
 	}
 }
 
-// IsRestartableSystemError checks if the error is caused by the system and
+// InvalidResourcesRequestError is an unrecoverable validation error from the underlying RM.
```

Comment:
validation of what?

Reply:
if someone tries to submit a job in a way that slurm rejects, we can catch it and prevent 5 retries https://github.com/determined-ai/determined-ee/pull/150/files#diff-8fffae72479c7eb5e7d37769d8b13e5867fae19ad9b63128bb727107c6b335d6R32

Reply:
trying to accomplish ^ with typed errors instead of regex'ing. also helps prevent rebase hell.